runtime/src/main/java/org/corfudb/protocols/wireprotocol/ReadResponse.java

```diff
@@ -19,20 +20,30 @@
     @Getter
     Map<Long, LogData> addresses;
 
+    @Getter
+    Map<UUID, Long> streamIdToCompactionMark;
+
     public ReadResponse(ByteBuf buf) {
         addresses = ICorfuPayload.mapFromBuffer(buf, Long.class, LogData.class);
+        streamIdToCompactionMark = ICorfuPayload.mapFromBuffer(buf, UUID.class, Long.class);
     }
 
     public ReadResponse() {
         addresses = new HashMap<Long, LogData>();
+        streamIdToCompactionMark = new HashMap<UUID, Long>();
     }
 
     public void put(Long address, LogData data) {
         addresses.put(address, data);
     }
 
+    public void putCompactionMark(UUID streamId, Long compactionMark) {
```

Comment:
It's easier to consume if the parameter is  Map<UUID, Long>, not separate.

Reply:
add setter